bluetooth: a2dp set right endpoint state for start and abort cmd

MarkWangChinese opened the pull request:
If the start cmd fail, need set back as OPEN state.

MarkWangChinese committed:
bluetooth: a2dp set right endpoint state for start and abort cmd

If the start cmd fail, need set back as OPEN state.

Signed-off-by: Mark Wang <yichang.wang@nxp.com>

lylezhu2012 reviewed (commented):
The change depends on the `req->status` is cleared by the upper layer.  I did not find the checking of `req->status` when calling AVDTP functions.
Add the owner of  `req` is AVDTP in function `avdtp_start_rsp()`, `avdtp_suspend_rsp()`, and `avdtp_abort_rsp()`. It is not reasonable for AVDTP to assume that the default value is `BT_AVDTP_SUCCESS`.

MarkWangChinese committed:
bluetooth: a2dp: set req->status as 0

create internal bt_a2dp_init_req and bt_a2dp_init_ctrl_req to init `req`,
then the `req->status` will not be forgotten to be cleared as 0.

Signed-off-by: Mark Wang <yichang.wang@nxp.com>

MarkWangChinese commented:
> The change depends on the `req->status` is cleared by the upper layer. I did not find the checking of `req->status` when calling AVDTP functions. Add the owner of `req` is AVDTP in function `avdtp_start_rsp()`, `avdtp_suspend_rsp()`, and `avdtp_abort_rsp()`. It is not reasonable for AVDTP to assume that the default value is `BT_AVDTP_SUCCESS`.

OK, add one commit to highlight that the `req->status` needs to be reset before giving `req` to avdtp.

MarkWangChinese committed:
bluetooth: a2dp: check the reject response's error code

If the reject response's error codes is success, change as FORMAT ERROR.

Signed-off-by: Mark Wang <yichang.wang@nxp.com>
